jupyter_book/sphinx.py

```diff
@@ -190,11 +190,16 @@ def debug_args():
     app = None  # In case we fail, this allows us to handle the exception
     try:
         # This patch is what Sphinx does, so we copy it blindly...
+        sourcedir_posix = sourcedir.as_posix()
```

Comment:
I'm a little bit unclear why this is necessary in jupyter-book. It should be _exactly_ copying what Sphinx does, so why would this work in Sphinx but not in Jupyter Book?

Reply:
Hi @choldgraf -- still some experimenting going on here but I wanted to make sure tests were passing on github.  On windows, there is only one failing test left:  test_toc_urllink, which is failing inside sphinx with a "Theme error:"
```
Theme error:
An error happened in rendering the page content1.
Reason: OSError(22, 'The filename, directory name, or volume label syntax is incorrect')
Traceback (most recent call last):
```
Any idea about where this rendering might be occurring, and why the code path is unique to urllink?  All
the other _toc variants build cleanly, the `jb page` tests are all working and the docs build. 

There is one other partial test failure: for the pdflatex builder the book.pdf is being built successfully in _build/pdf, but the builder reports a failure after the build completes.

I'm out of time  on this until next weekend, but if there was a release coming up, I think this is certainly usable on all the page and book content we're working with.

Reply:
hmmm - I am gonna try adding a PR that uses Sphinx's own `toctree` function instead of the pydata theme code, and maybe that will resolve this